packages/hash/frontend/src/components/GlideGlid/utils/interactable-manager.ts

```diff
@@ -0,0 +1,161 @@
+   * @param props Properties which will will be used to create the `Interactable`
+   * @returns The created `Interactable`
+   */
+  create(
+    args: DrawArgs<CustomCell>,
+    props: Omit<Interactable, "hovered" | "path" | "cellRect">,
+  ): Interactable {
+    // used -100 to prevent handling events on non-hovered cells
+    const { hoverX = -100, hoverY = -100, rect } = args;
+
+    const hovered = isCursorOnInteractable(
+      { posX: hoverX, posY: hoverY },
+      props.pos,
+      rect,
+    );
+
+    return {
+      ...props,
+      hovered,
+      cellRect: rect,
+      path: drawArgsToPath(args),
+    };
+  }
+
+  /**
+   * Call this function at the end of the `draw` function of the custom cell to register the interactables.
+   * @param args Draw args of cell
+   * @param interactables List of the interactables for a specific cell.
+   */
+  setInteractablesForCell(
+    args: DrawArgs<CustomCell>,
+    interactables: Interactable[],
+  ) {
+    const path = drawArgsToPath(args);
+
+    /**
+     * for each interactable, check if the hover status changed
+     * if it's changed, trigger corresponding event
+     */
+    for (const interactable of interactables) {
+      const existing = this.interactableStore[path]?.[interactable.id];
+
+      if (existing && existing.hovered !== interactable.hovered) {
+        const event = interactable.hovered
+          ? interactable.onMouseEnter
+          : interactable.onMouseLeave;
+
+        event?.(interactable);
+      }
+    }
+
+    const interactableMap = Object.fromEntries(
+      interactables.map((interactable) => [interactable.id, interactable]),
+    );
+
+    this.interactableStore[path] = interactableMap;
+  }
+
+  /**
+   * Checks if the cursor position is overlapping with an `Interactable` inside clicked cell.
+   * If it's overlapping, triggers the corresponding event handler for the clicked `Interactable`
+   * @returns true if handled the click event, false if not
+   */
+  handleClick(path: CellPath, event: CursorPos): boolean {
+    const interactableMap = this.interactableStore[path] ?? {};
+    const interactables = Object.values(interactableMap);
+
+    let foundInteractable: Interactable | undefined;
+
+    for (let i = 0; i < interactables.length; i++) {
+      const interactable = interactables[i]!;
+      const { pos, cellRect } = interactable;
+
+      const hovered = isCursorOnInteractable(event, pos, cellRect);
+
+      if (hovered) {
+        foundInteractable = interactable;
+      }
+    }
+
+    if (!foundInteractable) {
+      return false;
+    }
+
+    const { onClick, onMouseEnter } = foundInteractable;
+
+    const handler = onClick ?? onMouseEnter;
+
+    if (handler) {
+      handler?.(foundInteractable);
+
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * @param tableId Table to delete interactables
+   * @param boundaries If it's defined, function will delete `Interactables` saved for out-of-bound rows.
```

Comment:
Related to another comment – the out-of-boundaries rows will be deleted. This reads a bit like the out-of-boundary rows will be spared
```suggestion
   * @param boundaries If it's defined, function will delete `Interactables` before and after the boundaries
```